core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -105,7 +107,7 @@ public void onNoLongerMaster(String source) {
 
                 @Override
                 public void onFailure(String source, Exception e) {
-                    logger.error("unexpected failure during [{}]", e, source);
+                    logger.error((Supplier<?>) () -> new ParameterizedMessage("unexpected failure during [{}]", source), e);
```

Comment:
> But that's not any different than the current non-exception logging?

The whole point of going the `org.apache.log4j.util.Supplier<?>` route was to avoid creating a new instance of `ParameterizedMessage` if the log message is not going to be emitted. I don't think we want to go through this trouble to immediately turn around and trigger allocations from boxing and varargs (yes, the latter can be avoided with a lot of overrides but the former can not without adding silly overrides for every possible combination).

> log4j just creates a ton of overloads with Object parameters too to avoid the params, right?

Yes.